File: addons/point_of_sale/static/src/app/models/res_company.js
/** @odoo-module */
import { registry } from "@web/core/registry";
import { Base } from "./related_models";

export class ResCompany extends Base {
    static pythonModel = "res.company";
}

registry.category("pos_available_models").add(ResCompany.pythonModel, ResCompany);

File: addons/microsoft_outlook/models/res_company.py
from odoo import fields, models


class ResCompany(models.Model):
    _inherit = 'res.company'

    uses_microsoft_outlook_server = fields.Boolean('Use an Outlook Server', groups='base.group_system')
    microsoft_outlook_client_identifier = fields.Char('Outlook Client Id', groups='base.group_system')
    microsoft_outlook_client_secret = fields.Char('Outlook Client Secret', groups='base.group_system')
